### PR TITLE
fix: fall back on model support errors during auth rotation

### LINKED ISSUE #2221: bug: model-support 400/422 can cause alternating fallback failures
## Description
When a provider or pooled upstream returns a model-support error such as `400 invalid_request_error: The requested model is not supported.` the request can fail in an alternating pattern instead of falling through to another eligible auth or upstream.

## Root Cause
Two behaviors combine here:
- model-support `400` / `422` responses are treated like terminal request-shape errors, so fallback stops too early
- the failing auth-model path can remain immediately eligible for future round-robin selection, which makes hidden rotation show up as fail/succeed/fail/succeed

## Affected Files
- `sdk/cliproxy/auth/conductor.go`
- `sdk/cliproxy/auth/openai_compat_pool_test.go`
- `sdk/cliproxy/auth/conductor_overrides_test.go`

### PR DESCRIPTION
**Pull request by kaitranntt**
Closes #2221

## Summary
- treat model-support `400` / `422` responses as fallback-eligible instead of terminal request-shape failures
- suspend the failing auth-model path so round-robin does not immediately reselect it
- add regression coverage for pooled upstream fallback and cross-auth fallback

## Validation
- `go test ./sdk/cliproxy/auth/...`
- `go test ./sdk/cliproxy/...`
- `go test ./...`

### CODE REVIEW

**Comment by kaitranntt** [comment]
Pushed follow-up commit `ea3e0b71`.

This addresses both Codex review findings and also fixes one related stream edge case: pre-first-byte stream bootstrap failures now continue outer auth rotation when another auth is still available, while preserving the terminal error-stream contract when nothing else can run.

Verification run on this branch:
- `go test ./sdk/cliproxy/auth/...`
- `go test ./sdk/api/handlers/...`
- `go test ./sdk/cliproxy/...`
- `go test ./...`

File: sdk/cliproxy/auth/conductor.go [hunk 1]
```diff
@@ -1883,14 +1890,65 @@ func statusCodeFromResult(err *Error) int {
...
 // "invalid_request_error" and all 422 responses as request-shape failures,
-// where switching auths or pooled upstream models will not help.
+// where switching auths or pooled upstream models will not help. Model-support
+// errors are excluded so routing can fall through to another auth or upstream.
 func isRequestInvalidError(err error) bool {
 	if err == nil {
 		return false
 	}
+	if isModelSupportError(err) {
+		return false
```
[the remark this answers is not in the document]
Comment (kaitranntt) [reply sdk/cliproxy/auth/conductor.go:2067]: Addressed in ea3e0b71. `ExecuteCount` now records through `MarkResult`, so model-support fallback updates auth/model availability the same way as execute/stream. Added count-path regressions to cover repeated pooled requests after a model-support failure.

File: sdk/cliproxy/auth/conductor.go [hunk 2]
```diff
@@ -1627,53 +1627,60 @@ func (m *Manager) MarkResult(ctx context.Context, result Result) {
...
-					state.NextRetryAfter = next
-					suspendReason = "payment_required"
-					shouldSuspendModel = true
-				case 404:
+				if isModelSupportResultError(result.Error) {
 					next := now.Add(12 * time.Hour)
 					state.NextRetryAfter = next
-					suspendReason = "not_found"
+					suspendReason = "model_not_supported"
 					shouldSuspendModel = true
```
[the remark this answers is not in the document]
Comment (kaitranntt) [reply sdk/cliproxy/auth/conductor.go:1751]: Addressed in ea3e0b71. The fallback state is now keyed to the concrete upstream candidate for pooled aliases, later requests skip suspended upstream members, and a fully blocked pool auth no longer burns retry budget before a healthy fallback auth can run. Added execute/stream/count regressions for repeated pooled requests and the retry-budget edge case.
